### DISCUSSION TITLE
Does supabase require an S3-compatible storage? or is it built-in?

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [supabase/supabase#3707] [Questions] [answered]
Hi all, just a bit confused: does supabase provide the s3 service itself?

or does it require me to provide an S3-compatible credentials to an external s3 service?

Reason is I was thinking of using SeaweedFS as the s3 service https://github.com/chrislusf/seaweedfs

Thank you.

**Comment by soedirgo** [comment, marked as answer]
Hi Josh, if you're referring to the hosted service, Supabase does manage S3 by itself. The self-hosted version does let you provide your own S3 endpoint (it's backed by the filesystem by default), but I haven't tested this - we can provide better docs for this in the future.

**Reply by ecrfural** [reply]
is this made possible?

**Reply by haladynkamil** [reply]
I would also like to know how to connect my own S3 bucket to self hosted supabase

**Reply by brocococonut** [reply]
It'd be awesome if you could provide an external s3 endpoint for the managed service as well.

**Reply by mchtter** [reply]
You can use built-in if you want. optionally possible to use separately. if you want to use "AWS S3 Bucket" as an example;

.env settings:


TENANT_ID=blablabla
REGION={AWS_REGION}                                           <------ HERE IMPORTANT !
GLOBAL_S3_BUCKET={AWS_S3_NAME}                <------ HERE IMPORTANT !
POSTGREST_URL=http://localhost:3000
PGOPTIONS="-c search_path=storage,public"
FILE_SIZE_LIMIT=52428800
STORAGE_BACKEND=s3                                           <------ HERE IMPORTANT !
FILE_STORAGE_BACKEND_PATH=./data

**Reply by brocococonut** [reply]
I was meaning for the managed service, not the self-hosted one. Thanks though 😅

I've got an issue open on the underlying PHP framework that's been developed for a generalised SE storage provider that allows for custom endpoints. Currently you can only use AWS S3, Local, and I think backblaze.

**Reply by Shaun-Regenbaum** [reply]
I second this, I would really love to be able to provide my own s3 instance and be able to use supabase to handle metadata on hosted supbase.
